File: test_stats.py
#!/usr/bin/env python3
import json
import stats

# Load races and jobs
with open('races.json') as f:
    races = json.load(f)
with open('jobs.json') as f:
    jobs = json.load(f)

# Test character creation
elf_char = stats.create_character_stats(races, 'elf', 'archer', jobs, level=1)
print('=== Elf Archer Character ===')
print('Base Stats:', elf_char['base_stats'])
print('Level:', elf_char['level'])
print('Max HP:', elf_char['derived_stats'].get('max_hp'))
print('Max SP:', elf_char['derived_stats'].get('max_sp'))
print('ATK:', elf_char['derived_stats'].get('atk'))
print('MATK:', elf_char['derived_stats'].get('matk'))
print('DEF:', elf_char['derived_stats'].get('def'))
print('Flee:', elf_char['derived_stats'].get('flee'))
print('CRIT:', elf_char['derived_stats'].get('crit'), '%')
print()

print('=== Dwarf Blacksmith Character ===')
dwarf_char = stats.create_character_stats(races, 'dwarf', 'blacksmith', jobs, level=1)
print('Base Stats:', dwarf_char['base_stats'])
print('Max HP:', dwarf_char['derived_stats'].get('max_hp'))
print('ATK:', dwarf_char['derived_stats'].get('atk'))
print('DEF:', dwarf_char['derived_stats'].get('def'))
print('Weight Limit:', dwarf_char['derived_stats'].get('weight_limit'), 'kg')
print()

print('=== Demon Mage Character ===')
demon_char = stats.create_character_stats(races, 'demon', 'mage', jobs, level=1)
print('Base Stats:', demon_char['base_stats'])
print('Max HP:', demon_char['derived_stats'].get('max_hp'))
print('Max SP:', demon_char['derived_stats'].get('max_sp'))
print('MATK:', demon_char['derived_stats'].get('matk'))
print('MDEF:', demon_char['derived_stats'].get('mdef'))

print('\n✅ Stats system test completed successfully!')

File: src/test_stats.js
"use strict";

const path = require("path");
const fs = require("fs");
const stats = require("./stats");

const root = path.join(__dirname, "..");
const races = JSON.parse(fs.readFileSync(path.join(root, "races.json"), "utf8"));
const jobs = JSON.parse(fs.readFileSync(path.join(root, "jobs.json"), "utf8"));

const elfChar = stats.createCharacterStats(races, "elf", "archer", jobs, 1);
console.log("=== Elf Archer Character ===");
console.log("Base Stats:", elfChar.base_stats);
console.log("Level:", elfChar.level);
console.log("Max HP:", elfChar.derived_stats.max_hp);
console.log("Max SP:", elfChar.derived_stats.max_sp);
console.log("ATK:", elfChar.derived_stats.atk);
console.log("MATK:", elfChar.derived_stats.matk);
console.log("DEF:", elfChar.derived_stats.def);
console.log("Flee:", elfChar.derived_stats.flee);
console.log("CRIT:", elfChar.derived_stats.crit, "%");
console.log();

console.log("=== Dwarf Blacksmith Character ===");
const dwarfChar = stats.createCharacterStats(races, "dwarf", "blacksmith", jobs, 1);
console.log("Base Stats:", dwarfChar.base_stats);
console.log("Max HP:", dwarfChar.derived_stats.max_hp);
console.log("ATK:", dwarfChar.derived_stats.atk);
console.log("DEF:", dwarfChar.derived_stats.def);
console.log("Weight Limit:", dwarfChar.derived_stats.weight_limit, "kg");
console.log();

console.log("=== Demon Mage Character ===");
const demonChar = stats.createCharacterStats(races, "demon", "mage", jobs, 1);
console.log("Base Stats:", demonChar.base_stats);
console.log("Max HP:", demonChar.derived_stats.max_hp);
console.log("Max SP:", demonChar.derived_stats.max_sp);
console.log("MATK:", demonChar.derived_stats.matk);
console.log("MDEF:", demonChar.derived_stats.mdef);

console.log("\n✅ Stats system test completed successfully!");